crates/bevy_reflect/bevy_reflect_derive/src/derive_data.rs

```diff
@@ -147,8 +147,14 @@ impl<'a> ReflectDerive<'a> {
                     }
 
                     reflect_mode = Some(ReflectMode::Normal);
-                    let new_traits = ReflectTraits::from_nested_metas(&meta_list.nested)?;
-                    traits = traits.merge(new_traits)?;
+
+                    let mut metas = vec![];
+                    meta_list.parse_nested_meta(|meta| {
+                        metas.push(Meta::Path(meta.path));
+                        Ok(())
+                    })?;
+                    let new_traits = ReflectTraits::from_metas(metas)?;
+                    traits.merge(new_traits)?;
```

Comment:
By only passing in `Meta::Path`, aren't we preventing `Meta::List`-type attributes like `#[reflect( Hash(custom_hash_function) )]`?

We might need to just pass the whole `meta` (`ParseNestedMeta`) into `ReflectTraits::from_metas`.

Reply:
I... think not. I tried using the syntax in code and it parsed successfully, but I'm not sure enough on how to use it to be sure it works as expected

Reply:
Yeah just tested this out. This doesn't compile anymore:

```rust
#[derive(Reflect, Hash)]
#[reflect(Hash(custom_hash))]
struct Foo {
    bar: String,
}

fn custom_hash(x: &Foo) -> u64 {
    123
}
```

Reply:
thanks, should work now, I also added a test 👍